### PR TITLE
Bug 1988349: Insightsreport set corresponding clusteroperator condition correctly

### PR DESCRIPTION
**Pull request by tremes**
This PR registers the `insightsreport` status controller so that the operation can be distinguished and new condition (called `InsightsDownloadDegraded`) introduced 

## Categories

- [X] Bugfix
- [ ] Enhancement
- [ ] Backporting
- [ ] Others (CI, Infrastructure, Documentation)

## Sample Archive

No new data

## Documentation

No doc update

## Unit Tests

I will work on new integration test.

## Privacy

Yes. There are no sensitive data in the newly collected information.

## Breaking Changes

No

## References

https://issues.redhat.com/browse/???
https://bugzilla.redhat.com/show_bug.cgi?id=???
https://access.redhat.com/solutions/???

### CODE REVIEW

File: pkg/controller/status/status.go [hunk 1]
```diff
@@ -181,6 +188,11 @@ func (c *Controller) merge(existing *configv1.ClusterOperator) *configv1.Cluster
 				disabledReason = summary.Reason
 				disabledMessage = summary.Message
 			}
+		} else if summary.Operation == controllerstatus.DownloadingReport {
+			degradingFailure = false
+			klog.V(4).Info("Failed to download Insights report")
```
Comment (tremes) [inline pkg/controller/status/status.go:193]: I will try to think about some other/better message here....on the other hand it's pretty clear from the previous log messages that the download failed....

File: pkg/insights/insightsreport/insightsreport.go [hunk 2]
```diff
@@ -209,7 +219,7 @@ func (r *Gatherer) RetrieveReport() {
 }
 
 // Run goroutine code for gathering the reports from Smart Proxy
-func (r *Gatherer) Run(ctx context.Context) {
+func (r *Controller) Run(ctx context.Context) {
```
Comment (rluders) [inline pkg/insights/insightsreport/insightsreport.go:222]: `(c *Controller)`

**Review by rluders — CHANGES_REQUESTED** [review]

**Pull request merged**